src/messages/transaction.cpp

```diff
@@ -18,6 +20,13 @@ uint256 ComputeUTXOKey(const uint256& hash, uint32_t txIndex, uint32_t outIndex)
 /*
  * TxInput class START
  */
+
+TxInput::TxInput(const TxOutPoint& outpoint,
+                 const CPubKey& pubkey,
+                 const uint256& hashMsg,
+                 const std::vector<unsigned char>& sig)
+    : TxInput(outpoint, tasm::Listing{VStream(pubkey, sig, hashMsg)}) {}
+
```

Comment:
why move the constructor from .h file to .cpp file?

Reply:
in order to move `#include pubkey.h` from .h file to .cpp file